File: src/main/java/app/askresume/domain/job/model/JobMaster.java
package app.askresume.domain.job.model;

import app.askresume.domain.common.BaseTimeEntity;
import lombok.*;
import org.hibernate.annotations.*;

import javax.persistence.Entity;
import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Where(clause = "deleted_at = 'Y'")
@SQLDelete(sql = "UPDATE job_master SET deleted_at = 'N' WHERE id = ?")
@DynamicInsert
@DynamicUpdate
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class JobMaster extends BaseTimeEntity {

    @Builder.Default
    @OneToMany(mappedBy = "jobMaster", fetch = FetchType.LAZY)
    private List<Job> jobs = new ArrayList<>();

    @Comment(value = "작업마스터명")
    @Column(length = 150, nullable = false)
    private String masterName;

    @Comment(value = "삭제유무")
    @ColumnDefault("'Y'")
    @Column(length = 1, nullable = false)
    private String deletedAt;
}

File: src/main/kotlin/app/askresume/domain/job/model/JobMaster.kt
package app.askresume.domain.job.model

import app.askresume.domain.BaseEntity
import org.hibernate.annotations.Comment
import org.hibernate.annotations.SQLDelete
import org.hibernate.annotations.Where
import javax.persistence.*


@Where(clause = "is_deleted = false")
@SQLDelete(sql = "UPDATE job_master SET is_deleted = true WHERE id = ?")
@Entity
class JobMaster(
    masterName: String,

    @Comment(value = "id")
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    ) : BaseEntity() {

    @Comment(value = "작업마스터명")
    @Column(length = 150, nullable = false)
    var masterName: String = masterName
        protected set

}